### PR TITLE
feat(extract): add asyncapi import stage

### LINKED ISSUE #498: Add the AsyncAPI import stage
**Milestone:** Phase 13 Extraction Workflows — M2 (D2.2)
**PRD:** `docs/project/PRD/phase-13-extraction-workflows/PRD.md` §§4 and 6
**Architecture:** `docs/project/PRD/phase-13-extraction-workflows/ARCH.md` §3.6
**Dogfooding:** `docs/project/PRD/phase-13-extraction-workflows/dogfooding.md` §3.1 D0.2
**Delivery plan:** `docs/project/PRD/phase-13-extraction-workflows/delivery.md` D2.2

## Problem

Riviere extraction today is a sequence of manual CLI commands.

Teams that describe their broker messages and their publish and subscribe operations in an AsyncAPI document have to translate that document into the architecture graph by hand. The translation is a copy, so the graph carries whatever the person doing the copy understood, and there is nothing that compares the two afterwards. A message that is present in the AsyncAPI document and missing from the graph looks the same as a message that never existed.

This is one source of the wider problem in `problem-definition.md`: extracting a complete graph from a real system takes five or more coordinated steps, a failure can leave the graph complete, stale, or partly updated, and the workflow is not a dependable CI boundary.

## Solution

Add the `asyncapi-import` stage so `riviere-workflow.yaml` can read an AsyncAPI v3 document and add its messages and publish and subscribe operations to the graph, so the graph stops depending on a manual translation.

The `asyncapi-import` stage reads an AsyncAPI document and its mappings file, resolves each supported item to a Riviere component identity, and adds the components and Links to the graph the Workflow is building. Request and reply operations are not supported in Phase 13 and fail the stage. Servers and channels are not added as components. A message or operation that no mapping resolves is either a stage failure or a skipped record, according to `allow-unmapped` in `specs/asyncapi-import.yaml`.

This capability fits between the stages around it in `riviere-workflow.yaml`:

- The stage immediately before it is `eventcatalog-import`, which has already added components and Links from the EventCatalog source. `asyncapi-import` adds components and Links from the AsyncAPI document. Where both stages set the same scalar field on the same component, the later stage wins, per `PRD.md` §2.2 Sources of Truth First.
- The stage immediately after it is `ai-extract`, which reads the accumulated graph to find the gaps deterministic stages left. What passes between them is the accumulated graph inside `RiviereProject`'s private `RiviereBuilder`, and the immutable accumulated-state snapshot the Workflow records after each completed stage.
- This capability contributes the components and Links that only exist in the AsyncAPI document. `eventcatalog-import` does not read AsyncAPI, and `ai-extract` only fills gaps rather than importing a specification.

### Full architecture context

Read `docs/project/PRD/phase-13-extraction-workflows/ARCH.md` for the full approved architecture context and scope. This issue is not a replacement for `ARCH.md`.

### Target components, boundaries, and responsibilities

- `LoadAsyncApiDocument` is the Project's domain port for reading a resolved AsyncAPI document. `RiviereProject` depends on it. It is supplied through `RiviereProjectRepository`, not passed to `load` and not persisted.
- `AsyncApiDocumentAdapter` implements that port over `AsyncApiParserClient`. It translates parser results into the domain input and owns no scope, mapping, or graph decisions.
- `AsyncApiParserClient` is a generic external client in `infra/external-clients/asyncapi/`. It is the only location that imports `@asyncapi/parser`.
- `RiviereProject` owns the supported-scope decisions, the mapping from AsyncAPI items to canonical component identities, and Builder mutation.
- The supported scope is the exhaustive table in `PRD.md` §3.4 `#### asyncapi-import`. Any AsyncAPI v3 field not listed there is dropped without failing the stage.
- Component identity comes from `ComponentId.parseFromParts({ domain, module, type, name })`. A message becomes an `Event`; a mapped send or receive operation becomes the component type its mapping declares.

### Code/config/flow from ARCH.md

This is the whole of `specs/asyncapi-import.yaml`, a file the customer writes in their own repository. The `asyncapi-import` stage of `riviere-workflow.yaml` points at it.

```yaml
# asyncapi-import.yaml
source: ./asyncapi.yaml
mappings: ./asyncapi-mappings.yaml
allow-unmapped: false
```

This is part of `riviere-workflow.yaml`, the file the customer writes in the root of their own repository. The `stages` list is the ordered set of stages the workflow runs.

```yaml
- name: import-broker
  kind: asyncapi-import
  config: ./specs/asyncapi-import.yaml
```

This is the part of `ARCH.md` §3.6 that names the boundaries for this stage.

```text
The generic EventCatalog client imports only `@eventcatalog/sdk`; the generic AsyncAPI client imports only `@asyncapi/parser`. Their adapters implement the Project's ports. Project behaviour maps external records to canonical identities and mutates the Builder. Strict mode fails on unmapped records; lenient mode skips and records typed diagnostics/events.
```

### Role and location decisions

| Proposed Element | Kind | Role | Sublocation | Change | Confidence | Notes |
| --- | --- | --- | --- | --- | --- | --- |
| LoadAsyncApiDocument | type/interface | `domain-port` | extract domain-model domain/ports/ | new | High | Domain-owned resolved-document contract |
| AsyncApiDocumentAdapter | function/class | `domain-port-adapter` | extract use-cases features/extract/adapters/ | new | High | Translates parser client results |
| AsyncApiParserClient | function/class | `external-client-service` | extract use-cases infra/external-clients/asyncapi/ | new | High | Only location importing the parser |
| RiviereProject | class | `aggregate` | extract domain-model domain/ | changed | High | Owns scope, mapping, and mutation |

### Structural concerns and forbidden alternatives

- Do not import the parser outside its generic client.
- Do not add channels or servers as components.
- Do not fail the stage for an unrecognised top-level AsyncAPI field.

### Design challenge instruction

> If implementation reveals the agreed design is impractical, incomplete, or sub-optimal, stop and push back. Do not silently implement a different design.

### Product

**Rule:** AsyncAPI messages are added as `Event` components in a Riviere graph.

**Example:** `OrderPlacedMessage` is added as the `OrderPlaced` event.

Given the AsyncAPI document `specs/asyncapi.yaml`

```yaml
asyncapi: 3.0.0
info:
  title: Ecommerce broker
  version: 1.0.0
channels:
  orderPlaced:
    address: orders.placed
    messages:
      orderPlaced:
        $ref: '#/components/messages/OrderPlacedMessage'
operations:
  processOrder:
    action: send
    channel:
      $ref: '#/channels/orderPlaced'
components:
  messages:
    OrderPlacedMessage:
      name: OrderPlacedMessage
      payload:
        type: object
```

And the configuration file `specs/asyncapi-import.yaml`

```yaml
source: ./asyncapi.yaml
mappings: ./asyncapi-mappings.yaml
allow-unmapped: false
```

And the configuration file `specs/asyncapi-mappings.yaml`

```yaml
messages:
  OrderPlacedMessage:
    domain: orders-domain
    module: infrastructure
    name: OrderPlaced
operations:
  processOrder:
    type: UseCase
    domain: orders-domain
    module: consumer/order-placed
    name: ProcessOrder
```

When running the `asyncapi-import` stage of the workflow

Then the following items are added to the Riviere graph

```json
{
  "id": "orders-domain:infrastructure:event:orderplaced",
  "type": "Event",
  "name": "OrderPlaced",
  "eventName": "OrderPlaced",
  "domain": "orders-domain",
  "module": "infrastructure",
  "sourceLocation": {
    "repository": "https://github.com/ntcoding/ecommerce-demo-app",
    "filePath": "specs/asyncapi.yaml"
  }
}
```

```json
{
  "id": "orders-domain:consumer/order-placed:usecase:processorder",
  "type": "UseCase",
  "name": "ProcessOrder",
  "domain": "orders-domain",
  "module": "consumer/order-placed",
  "sourceLocation": {
    "repository": "https://github.com/ntcoding/ecommerce-demo-app",
    "filePath": "specs/asyncapi.yaml"
  }
}
```

**Rule:** An AsyncAPI send operation links its mapped component to the message's component, with type `async`.

**Example:** The `processOrder` operation sends `OrderPlacedMessage`.

Given the AsyncAPI document and configuration files shown in the example above

When running the `asyncapi-import` stage of the workflow

Then the following item is added to the Riviere graph

```json
{
  "source": "orders-domain:consumer/order-placed:usecase:processorder",
  "target": "orders-domain:infrastructure:event:orderplaced",
  "type": "async"
}
```

**Rule:** An AsyncAPI receive operation links the message's component to the operation's mapped component, with type `async`.

**Example:** The `handleOrderPlaced` operation receives `OrderPlacedMessage`.

Given the AsyncAPI document `specs/asyncapi.yaml` and the configuration files shown in the example above, with the `operations` mapping below

```yaml
operations:
  processOrder:
    type: UseCase
    domain: orders-domain
    module: consumer/order-placed
    name: ProcessOrder
  handleOrderPlaced:
    type: EventHandler
    domain: shipping-domain
    module: consumer/order-confirmed
    name: HandleOrderPlaced
```

When running the `asyncapi-import` stage of the workflow

Then the following item is added to the Riviere graph

```json
{
  "source": "orders-domain:infrastructure:event:orderplaced",
  "target": "shipping-domain:consumer/order-confirmed:eventhandler:handleorderplaced",
  "type": "async"
}
```

**Rule:** A request and reply operation fails the stage.

**Example:** The `processOrder` operation has a `reply` field.

Given the AsyncAPI document `specs/asyncapi.yaml` contains an operation with a `reply` field

When running the `asyncapi-import` stage of the workflow

Then the stage fails with

```text
asyncapi request/reply pattern not supported in Phase 13 (operation: 'processOrder')
```

and `.riviere/ecommerce-architecture.json` is unchanged.

**Rule:** A message or operation with no mapping fails the stage when `allow-unmapped` is `false`.

**Example:** `OrderPlacedMessage` has no mapping and `allow-unmapped` is `false`.

Given the AsyncAPI document `specs/asyncapi.yaml` and the configuration file

```yaml
# specs/asyncapi-import.yaml
source: ./asyncapi.yaml
mappings: ./asyncapi-mappings.yaml
allow-unmapped: false
```

And the configuration file

```yaml
# specs/asyncapi-mappings.yaml
messages: {}
operations: {}
```

When running the `asyncapi-import` stage of the workflow

Then the stage fails and `.riviere/ecommerce-architecture.json` is unchanged.

**Rule:** A message or operation with no mapping is skipped and recorded as a diagnostic when `allow-unmapped` is `true`.

**Example:** `OrderPlacedMessage` has no mapping and `allow-unmapped` is `true`.

Given the AsyncAPI document `specs/asyncapi.yaml` and the configuration file

```yaml
# specs/asyncapi-import.yaml
source: ./asyncapi.yaml
mappings: ./asyncapi-mappings.yaml
allow-unmapped: true
```

And the configuration file

```yaml
# specs/asyncapi-mappings.yaml
messages: {}
operations: {}
```

When running the `asyncapi-import` stage of the workflow

Then the stage completes and the run result contains an `unmapped-record` diagnostic naming `OrderPlacedMessage`.

### Design

- `specs/asyncapi-import.yaml` and `specs/asyncapi-mappings.yaml` are validated by JSON Schema before the typed stage is constructed, following the same rules as the EventCatalog schemas.
- Channels and servers produce no components.
- The AsyncAPI v3 scope is the exhaustive table in `PRD.md` §3.4 `#### asyncapi-import`, and it is implemented as written. Every field marked `Dropped` is dropped without failing the stage.
- No import of `@asyncapi/parser` exists outside `AsyncApiParserClient`.

### Quality

- Tests cover adapter translation, mappings, the documented scope, the unsupported request and reply pattern, scalar overwrite, Link occurrence, and the demo fixture.
- The adapter test uses a mocked generic client. One integration test parses the demo fixture with the real parser.

### Change areas

- `packages/riviere-extract-config/published-language`: AsyncAPI import and mappings schemas and published types
- `packages/riviere-extract-ts/domain-model`: `LoadAsyncApiDocument` port, stage behaviour, and `RiviereProject` wiring
- `packages/riviere-extract-ts/use-cases`: `AsyncApiParserClient`, `AsyncApiDocumentAdapter`, and repository collaborator supply
- `packages/riviere-extract-ts/domain-model` tests: stage behaviour, scope, and mapping tests

### Relevant repository rules

- `docs/conventions/software-design.md` SD-001 (fail fast), SD-002/SD-003 (exhaustive type-safe states), SD-004 (dependency injection), SD-005 (intention-revealing names), and SD-021 (YAGNI) apply where this ticket touches code.
- `docs/conventions/testing.md` TS-001 through TS-004 require outcome-named tests with exact value assertions and one concept per test; TS-008/TS-014 require relevant boundary cases.

### Relevant architecture memories

- `project-memory/architecture/memories/repository-loading-must-load-full-aggregate.md`: repository criteria describe full aggregate state, not the operation that follows loading.
- `project-memory/architecture/memories/ddd-modelling-trade-offs-from-riviere-workflows.md`: Project owns empty-start rebuild meaning and Workflow owns the stage lifecycle; repositories and use cases do not become the engine.

### Lint, role-check, and test expectations

Use role and lint failures to improve the design; do not add ignores, exclusions, assertions, optional production dependencies, or coverage workarounds. Preserve exhaustive unions and avoid the TypeScript `in` operator.

## Dependencies

- #492 (D0.2)
- #493 (D1.1)
- #494 (D1.2)
- #496 (D1.5)

## How to verify

```bash
pnpm nx test riviere-extract-ts-domain-model -- --testNamePattern "asyncapi"
```

Expected: the stage tests covering messages, send, receive, the request and reply failure, and strict and lenient outcomes pass with exact value assertions.

```bash
pnpm nx test riviere-extract-config-published-language
```

Expected: the AsyncAPI import and mappings schema tests pass.

```bash
pnpm verify
```

Expected: all targets pass.

## Things this ticket must not do

- Do not model request and reply operations.
- Do not add channels or servers as components.
- Do not map external participants or broader AsyncAPI semantics.
- Do not put scope or mapping decisions in the adapter.

## Glossary

Source: `docs/architecture/domain-terminology/contextive/definitions.glossary.yml`, and `docs/project/PRD/active/PRD-phase-13-extraction-workflows.md` §12.

**Component**: A discrete unit of software functionality in a Riviere graph. Has a type, belongs to a domain, and connects to other components via links.

**Link**: A directed relationship from one source Component to one target Component. Each Link represents one source occurrence and may have a Relationship Type, condition, sync or async type, and Source Location.

**Domain**: A logical grouping of related components. Each component belongs to exactly one domain. Domains have a systemType (domain, bff, ui, external-service, other).

**UseCase**: An application service component orchestrating business logic. Coordinates between UI/API layer and domain operations.

**Event**: A published event component representing something that happened in the domain. Has an eventName property.

**EventHandler**: A component that subscribes to and processes events. Has a subscribedEvents property listing event names it handles.

**Workflow**: An aggregate entity owned by `RiviereProject`. It has identity and run state and owns ordered stage progression, fail-fast behaviour, events, warnings, and diagnostics. A validated YAML definition is one persisted representation loaded by the Project repository.

**Workflow Stage**: An immutable value object in the owning Workflow's closed ordered stage union. The Project exhaustively performs the stage's domain behaviour while the Workflow owns progression.

**Stage Kind**: One of `code-extraction`, `eventcatalog-import`, `asyncapi-import`, `ai-extract`, `ai-enrich`, or `schema-validate`.

**Stage Config**: Validated typed configuration specific to a stage kind, stored in a separate file referenced by the workflow representation and resolved by `RiviereProjectRepository`.

**Project Domain Port**: An explicit capability needed by Project behaviour, such as loading EventCatalog/AsyncAPI data or invoking an AI CLI. A use-case adapter implements one port over one generic external client and contains no domain decisions.

**Mappings File**: Configuration defining how external data models (EventCatalog, AsyncAPI) map to Riviere concepts. Convention-based defaults with explicit overrides.

**Canonical Identity**: The final Riviere component identity produced by stage mappings/config and Project behaviour before the Builder sees it. Upsert happens only after identity is established.

**Workflow Diagnostics**: Workflow-owned state for unresolved draft-only diagnostics. Project stage operations report outcomes to the Workflow, which records and resolves diagnostics and emits events.

**Upsert**: Typed builder capability (one `upsert*` method per component type) to add-or-merge a component. If the component ID already exists, scalar fields are merged **last-wins** by default (or preserved under `{ noOverwrite: true }`) and array fields union. If not, it creates the component. Enables multi-source graph construction.

### PR DESCRIPTION
**Pull request by NTCoding**
## Description

Implements issue #498 by adding the AsyncAPI import workflow stage. It parses AsyncAPI v3 documents through the generic client, materializes configured message and operation mappings, creates canonical graph components and async links, and reports unsupported or unmapped records with the required typed behavior.

## Linked Issue

Closes #498

## What Problem Does This PR Solve?

Extraction workflows could configure an AsyncAPI stage but had no parser-backed import implementation, so AsyncAPI documents could not contribute messages, operations, or asynchronous links to the graph.

## Acceptance Criteria

Import AsyncAPI v3 send and receive operations only; map messages to Events and operations to configured component types; create correctly directed async links; reject request/reply; support strict and lenient unmapped records; use @asyncapi/parser only from the generic client.

## Key Changes

Added AsyncAPI mapping schema/parser, LoadAsyncApiDocument port, domain import stage, parser client and adapter, CLI composition wiring, repository materialization, fixture-based parser integration coverage, and graph serialization external client extraction.

## Notable Architectural Changes / Impact

Adds the AsyncAPI adapter boundary and domain port; no role exceptions. Graph serialization is now an external-client service, leaving aggregate loading in RiviereProjectRepository.

## Validation

Pre-commit full verification passed: lint-md, role-check, build, depcruise-eclair, lint, typecheck, test, generated-doc checks, and knip. CLI: 691 tests with 100% coverage; domain model: 449 tests with 100% coverage.

## Notes

@asyncapi/parser 3.6.3 is scoped to riviere-extract-ts-use-cases.

## Summary by CodeRabbit

- **New Features**
  - Added support for importing AsyncAPI 3 documents into project graphs.
  - AsyncAPI messages and operations can now be mapped to project components and relationships.
  - Added support for inline AsyncAPI mapping configuration, including strict or lenient handling of unmapped records.
  - Added validation and clear error reporting for invalid mappings, unsupported operations, and unreadable documents.
  - Added duplicate-safe link creation to prevent repeated relationships.

- **Bug Fixes**
  - EventCatalog imports now handle duplicate links consistently without creating duplicates.

## Approved scope amendment

The user approved extending this PR to repair dev-workflow-v2 state instructions after the workflow entered REVIEWING with an unnamed state-read action and the agent invoked an unsupported operation. The amendment removes redundant agent-side workflow-state checks, makes remaining workflow actions explicit, moves review-cycle bookkeeping and the cap notice into workflow-owned operations, and corrects stale IMPLEMENTING-to-REVIEWING handoffs. This is approved because the workflow defect blocked the PR review process.

### CODE REVIEW

File: packages/riviere-extract-ts/use-cases/src/infra/external-clients/filesystem/serialized-graph-writer.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { mkdirSync, writeFileSync } from 'node:fs'
+import { dirname } from 'node:path'
+
+/** @riviere-role external-client-service */
+export function writeSerializedGraph(graphFileLocation: string, graph: string): void {
```
Comment (NTCoding) [inline packages/riviere-extract-ts/use-cases/src/infra/external-clients/filesystem/serialized-graph-writer.ts]: FAIL: graph is a domain concept, this is an external client. This component was not approved in the plan or anywhere.
